plugins/plugin_utils/terminalmanager.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: Contributors to the Ansible project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+import random
+import re
+import string
+
+from ansible.module_utils._text import to_bytes
+from ansible.module_utils._text import to_text
+from ansible.plugins.shell.powershell import _common_args
+
+
+class TerminalManager:
+    def __init__(self, connection):
+        self.connection = connection
```

Comment:
```suggestion
    def __init__(self, session: Any, stdout: Any, poller: Callable, verbosity_display: Callable) -> None:
        self._session = session
        self._stdout = stdout
        self._poller = poller
        self.verbosity_display = verbosity_display
```
Having the connection as a parameter does not make the class reusable; we should pass only the necessary options needed by the class